### PR TITLE
add pattern blame

### PR DESCRIPTION
**Pull request by Glease**
so you know who was that idiot adding broken pattern
Closes https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/14297

image

### CODE REVIEW

**Comment by Rika-TH** [comment]
Perfection, keep whoever makes the bad ones in charge of AE to get updoots 👍

**Comment by JodieRuth** [comment]
Kick this idiot out of the server
image

File: src/main/resources/assets/appliedenergistics2/lang/en_US.lang [hunk 1]
```diff
@@ -202,6 +202,7 @@ gui.appliedenergistics2.Substitute=Using Substitutions:
 gui.appliedenergistics2.BeSubstitute=Can be a substitution:
 gui.appliedenergistics2.Yes=Yes
 gui.appliedenergistics2.No=No
+gui.appliedenergistics2.AuthoredBy=Authored by: %s
```
Comment (firenoo) [inline src/main/resources/assets/appliedenergistics2/lang/en_US.lang:205]: Would suggest: "Encoder:", "Encoded by:", etc as Authored By makes it sound like a git commit

**Review by firenoo — DISMISSED** [review]
Resolve regardless of action taken before merging

File: src/main/resources/assets/appliedenergistics2/lang/en_US.lang (hunk 1 above)
Comment (Glease) [reply src/main/resources/assets/appliedenergistics2/lang/en_US.lang:205]: Encoded by sounds better

**Comment by Glease** [comment]
we have discovered a subtle problem with the implementation. if there were two different people encoded the same recipe, they will be treated as two different pattern, and AE won't load balance between the two. I have come up with [a solution of preprocessing the item used for equality check in PatternHelper](https://github.com/GTNewHorizons/Applied-Energistics-2-Unofficial/blob/5428971041e9b9736537a440b67252e1bb4099dd/src/main/java/appeng/helpers/PatternHelper.java#L70-L76), but I'm not sure if this is worth it (as copying an NBT can be expensive)

**Comment by Glease** [comment]
well AE2 has a shared NBT cache, so it'd actually be faster to compare full stack (which is just a == check between the two NBT tag) than to compare by indiviual inputs (which will need to iterate over every inputs and go into every subtag the inputs/outputs have)

**Comment by firenoo** [comment]
PatternHelper usages needs to be cleaned up anyways as its instantiated way too often so I think its fine the way you implemented it (with those patches)

**Review by firenoo — APPROVED** [review]

**Comment by Glease** [comment]
we will see how much of a performance impact this has

**Pull request merged**